### DISCUSSION TITLE
GetMetricConfiguration as a stream?

### DISCUSSION THREAD

**Discussion by lebogg** [clouditor/clouditor#533] [Q&A] [answered]
Currently, the `MetricConfigurationSource` implementation uses  `GetMetriConfiguration` which is a unary gRPC. Would it make sense to add a streaming version?

The question came to my mind as I ran the engine and saw that there are several `GetMetricConfiguration` calls. I do not know at which threshold (if there is any) it is wise to move to streaming instead of doing unary calls.

We may also use `ListMetricConfigurations` get all configuration per servide id at once?

@oxisto

**Comment by oxisto** [comment, marked as answer]
My suggestion is to rework this entirely. With #775 we finally have the metric configurations in the database and we can access the cloud service specific ones directly in the `CloudService` object. With this in place my suggestion is that the assessment fetches the override metric configurations using the `GetCloudService` call. For those metrics that are not override, a single call to a possibly new method `ListDefaultMetricConfigurations` should suffice (for the lifetime of the service), since the default does not change.
